x-pack/plugins/observability/public/hooks/slo/use_create_slo.ts

```diff
@@ -5,75 +5,91 @@
+      return http.post<CreateSLOResponse>(`/api/observability/slos`, { body });
     },
-    [http]
+    {
+      mutationKey: ['createSlo'],
+      onSuccess: () => {
+        queryClient.invalidateQueries(['fetchSloList']);
+        queryClient.invalidateQueries(['fetchHistoricalSummary']);
+      },
+    }
   );
 
-  const updateSlo = useCallback(
-    async (sloId: string, slo: UpdateSLOInput) => {
-      setLoading(true);
-      setError('');
-      setSuccess(false);
+  const cloneSlo = useMutation<
+    CreateSLOResponse,
+    string,
+    { slo: CreateSLOInput; idToCopyFrom?: string },
+    { previousSloList: FindSLOResponse | undefined }
+  >(
+    ['cloneSlo'],
+    ({ slo }: { slo: CreateSLOInput; idToCopyFrom?: string }) => {
       const body = JSON.stringify(slo);
+      return http.post<CreateSLOResponse>(`/api/observability/slos`, { body });
+    },
+    {
+      onMutate: async ({ slo, idToCopyFrom }) => {
+        // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+        await queryClient.cancelQueries(['fetchSloList']);
+
+        // Snapshot the previous value
+        const previousSloList = queryClient.getQueryData<FindSLOResponse>(['fetchSloList']);
 
-      try {
-        await http.put<UpdateSLOResponse>(`/api/observability/slos/${sloId}`, { body });
-        setSuccess(true);
-      } catch (e) {
-        setError(e);
-      } finally {
-        setSuccess(false);
-        setLoading(false);
-      }
+        const sloUsedToClone = previousSloList?.results.find((el) => el.id === idToCopyFrom);
+
+        // Optimistically update to the new value
+        queryClient.setQueryData(['fetchSloList'], () => ({
+          ...previousSloList,
+          results: [...(previousSloList?.results || []), { ...sloUsedToClone, name: slo.name }],
+        }));
+
+        // Return a context object with the snapshotted value
+        return { previousSloList };
+      },
+      // If the mutation fails, use the context returned from onMutate to roll back
+      onError: (_err, _slo, context) => {
+        if (context?.previousSloList) {
+          queryClient.setQueryData(['fetchSloList'], context.previousSloList);
+        }
+      },
+      onSuccess: () => {
+        queryClient.invalidateQueries(['fetchSloList']);
+        queryClient.invalidateQueries(['fetchHistoricalSummary']);
+      },
+    }
+  );
+
+  const updateSlo = useMutation(
+    ({ sloId, slo }: { sloId: string; slo: UpdateSLOInput }) => {
+      const body = JSON.stringify(slo);
+      return http.put<UpdateSLOResponse>(`/api/observability/slos/${sloId}`, { body });
     },
-    [http]
+    {
+      mutationKey: ['updateSlo'],
+      onSuccess: () => {
+        queryClient.invalidateQueries(['fetchSloList']);
+        queryClient.invalidateQueries(['fetchHistoricalSummary']);
+      },
+    }
   );
 
-  return {
-    loading,
-    error,
-    success,
-    createSlo,
-    updateSlo,
-  };
+  return { createSlo, cloneSlo, updateSlo };
```

Comment:
All these objects have their own `isLoading`, `isError`, `isSuccess`, `error`, `mutate`, `mutateAsync`, etc. inside them so you have more granular control about which request succeeded or failed.